Controller/MediaAdminController.php

```diff
@@ -109,6 +117,132 @@ public function listAction(Request $request = null)
     }
 
     /**
+     * @param Request $request
+     *
+     * @throws \Symfony\Component\Security\Core\Exception\AccessDeniedException|\LogicException
+     *
+     * @return \Symfony\Bundle\FrameworkBundle\Controller\Response|\Symfony\Component\HttpFoundation\Response
+     */
+    final public function multiUploadAction(Request $request)
+    {
+        $this->admin->checkAccess('create');
+
+        $providerName = $this->getRequest()->get('provider');
+        $provider = !empty($providerName) ? $this->get($providerName) : null;
+
+        $defaultContext = $this->get('sonata.media.pool')->getDefaultContext();
+        $context = $request->get('context', $defaultContext);
+
+        if (!$provider) {
+            $pool = $this->get('sonata.media.pool');
+            /** @var $providers MediaProviderInterface[] */
+            $providers = $pool->getProvidersByContext($context);
+
+            $filteredProviders = array();
+            foreach ($providers as $provider) {
```

Comment:
Please remove the `$providers` OTV and directly write: `foreach ($pool->getProvidersByContext($context) as $provider)`